tests/codecs/test_json.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+
+from typing import Any, Union
+
+from mlserver.codecs.json import decode_from_bytelike_json_to_dict, encode_to_json_bytes
+
+
+@pytest.mark.parametrize(
+    "input, expected",
+    [
+        (b"{}", dict()),
+        ("{}", dict()),
+        ('{"hello":"world"}', {"hello": "world"}),
+        (b'{"hello":"world"}', {"hello": "world"}),
+        (b'{"hello":"' + "world".encode("utf-8") + b'"}', {"hello": "world"}),
+        (
+            b'{"hello":"' + "world".encode("utf-8") + b'", "foo": { "bar": "baz" } }',
+            {"hello": "world", "foo": {"bar": "baz"}},
+        ),
+    ],
+)
+def test_decode_input(input: Union[str, bytes], expected):
+    assert expected == decode_from_bytelike_json_to_dict(input)
+
+
+@pytest.mark.parametrize(
+    "expected, input",
```

Comment:
Nit: I got initially confused until I noticed that `expected` is the first parameter. At least add a note for the reader.